src/sync-client.js

```diff
@@ -5,1157 +5,1178 @@ var cidProvider = require('./clientIdProvider');
-    "file_system_quota" : 50 * 1024 * 1204,
-    // Amount of space to request from the HTML5 filesystem API when running in browser
-    "icloud_backup" : false, //ios only. If set to true, the file will be backed by icloud
-    // If set, the client will resend those inflight pending changes that have lived longer than this value.
-    // This is to prevent the situation where updates are lost for certain pending changes, those pending changes will be stuck on the client forever.
-    // Default value is 24 hours.
-    "resend_inflight_pendings_minutes": 60*24
-  },
-
-  notifications: {
-    "CLIENT_STORAGE_FAILED": "client_storage_failed",
-    // loading/saving to client storage failed
-    "SYNC_STARTED": "sync_started",
-    // A sync cycle with the server has been started
-    "SYNC_COMPLETE": "sync_complete",
-    // A sync cycle with the server has been completed
-    "OFFLINE_UPDATE": "offline_update",
-    // An attempt was made to update a record while offline
-    "COLLISION_DETECTED": "collision_detected",
-    //Update Failed due to data collision
-    "REMOTE_UPDATE_FAILED": "remote_update_failed",
-    // Update Failed for a reason other than data collision
-    "REMOTE_UPDATE_APPLIED": "remote_update_applied",
-    // An update was applied to the remote data store
-    "LOCAL_UPDATE_APPLIED": "local_update_applied",
-    // An update was applied to the local data store
-    "DELTA_RECEIVED": "delta_received",
-    // A delta was received from the remote data store for the dataset 
-    "RECORD_DELTA_RECEIVED": "record_delta_received",
-    // A delta was received from the remote data store for the record 
-    "SYNC_FAILED": "sync_failed"
-    // Sync loop failed to complete
-  },
-
-  datasets: {},
-
-  // Initialise config to default values;
-  config: undefined,
-
-  //TODO: deprecate this
-  notify_callback: undefined,
-
-  notify_callback_map : {},
-
-  init_is_called: false,
-
-  //this is used to map the temp data uid (created on client) to the real uid (created in the cloud)
-  uid_map: {},
-
-  // PUBLIC FUNCTION IMPLEMENTATIONS
-  init: function(options) {
-    self.consoleLog('sync - init called');
-
-    self.config = JSON.parse(JSON.stringify(self.defaults));
-    for (var i in options) {
-      self.config[i] = options[i];
-    }
-
-    //prevent multiple monitors from created if init is called multiple times
-    if(!self.init_is_called){
-      self.init_is_called = true;
-      self.datasetMonitor();
-    }
-    defaultCloudHandler.init(self.config.cloudUrl, self.config.cloudPath);
-  },
-
-  notify: function(datasetId, callback) {
-    if(arguments.length === 1 && typeof datasetId === 'function'){
-      self.notify_callback = datasetId;
-    } else {
-      self.notify_callback_map[datasetId] = callback;
-    }
-  },
-
-  manage: function(dataset_id, opts, query_params, meta_data, cb) {
-    self.consoleLog('manage - START');
+module.exports = newClient;
+
+function newClient(id) {
```

Comment:
99% of the changes to this file are indenting the extra level due to everything being wrapped in this `newClient(id)` function.